modules/representation_management/spec/models/representation_management/accredited_entity_query_spec.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe RepresentationManagement::AccreditedEntityQuery, type: :model do
+  describe '#results' do
+    it 'returns nothing for a blank query string' do
+      expect(described_class.new('').results).to be_empty
+    end
+
+    it 'returns individuals and organizations in the sorted order' do
+      create(:accredited_individual, :with_location, full_name: 'Bob Law')
+      create(:accredited_organization, :with_location, name: 'Bob Law Firm')
+      create(:accredited_individual, :with_location, full_name: 'Bob Smith')
+      create(:accredited_organization, :with_location, name: 'Bob Smith Firm')
+
+      results = described_class.new('Bob').results
+
+      expect(results.size).to eq(4)
+      expect(results.first).to be_a(AccreditedIndividual)
+      expect(results.first.full_name).to eq('Bob Law')
+      expect(results.second).to be_a(AccreditedIndividual)
+      expect(results.second.full_name).to eq('Bob Smith')
+      expect(results.third).to be_a(AccreditedOrganization)
+      expect(results.third.name).to eq('Bob Law Firm')
+      expect(results.last).to be_a(AccreditedOrganization)
+      expect(results.last.name).to eq('Bob Smith Firm')
+    end
+
+    it 'sorts individuals by levenshtein distance' do
+      create(:accredited_individual, :with_location, full_name: 'aaaabc')
+      create(:accredited_individual, :with_location, full_name: 'aaaab')
+      create(:accredited_individual, :with_location, full_name: 'aaaabcde')
+      create(:accredited_individual, :with_location, full_name: 'aaaa')
+      create(:accredited_individual, :with_location, full_name: 'aaaabcd')
+
+      results = described_class.new('aaaa').results
+
+      expect(results.map(&:full_name)).to eq(%w[aaaa aaaab aaaabc aaaabcd aaaabcde])
+    end
+
+    it 'sorts organizations by levenshtein distance' do
+      create(:accredited_organization, :with_location, name: 'aaaabc')
+      create(:accredited_organization, :with_location, name: 'aaaab')
+      create(:accredited_organization, :with_location, name: 'aaaabcde')
+      create(:accredited_organization, :with_location, name: 'aaaa')
+      create(:accredited_organization, :with_location, name: 'aaaabcd')
+
+      results = described_class.new('aaaa').results
+
+      expect(results.map(&:name)).to eq(%w[aaaa aaaab aaaabc aaaabcd aaaabcde])
+    end
+
+    it 'sorts individuals and organizations together by levenshtein distance' do
+      create(:accredited_individual, :with_location, full_name: 'aaaabc')
+      create(:accredited_individual, :with_location, full_name: 'aaaab')
+      create(:accredited_organization, :with_location, name: 'aaaabcde')
+      create(:accredited_organization, :with_location, name: 'aaaa')
+      create(:accredited_individual, :with_location, full_name: 'aaaabcd')
+
+      results = described_class.new('aaaa').results
+
+      expect(results.map { |r| r.is_a?(AccreditedIndividual) ? r.full_name : r.name })
```

Comment:
- If you redesign the test data setup, then you could pluck the ids and check their order to clean this logic up slightly.
  - Or, you could keep this setup and assign the creates to variables to pluck and then check something like `[ind1.id, ind2.id, org1.id, ...]`

Reply:
I ended up going in this direction.